azkaban-exec-server/src/test/java/azkaban/execapp/FlowRunnerTestBase.java

```diff
@@ -170,4 +171,8 @@ protected void succeedJobs(final String... jobs) {
     }
   }
 
+  protected void assertEvents(final Type... types) {
```

Comment:
> optimize for reading than writing

Maybe I wasn't clear enough, but I didn't really mean that "easy writing" would mean "less writing". I was only aiming for maximal clarity.

I actually feel that my way here was about _optimizing for reading_: that you don't have a lot of noise in the code, just the essential information in the context.

For this kind of unit test where we test multiple different scenarios and workflows on the same class, I think that less noise is more important than avoiding redirection or inheritance.

I guess my thinking there is that you already know for example that `assertEvents` will check against the "current `eventCollector`" or that `assertStatus` will check against the "current flow" withouth having to every time echo the reference to event collector or flow runner.